feat: add api health check service for v4 (#3077)
gravitee-io/gravitee-api-management: gravitee-apim-plugin/gravitee-apim-plugin-apiservice/gravitee-apim-plugin-apiservice-healthcheck-http/README.adoc

= Http health-check service

== Description

> The *Http health-check Service* can be used to test if a backend service is available. If the service is considered as down by the health-check, it will be removed from the list of service so the loadbalancer will ignore this endpoint until the health-check will consider it again as available.

Reviewer: ```suggestion
The *Http health-check Service* can be used to test if a backend service is available. If the service is considered down by the health-check, it will be removed from the list of services, so the load balancer will ignore this endpoint until the health-check considers it again as available.
```

> The *Http health-check Service* checkes the heathly state of the backend service by sending to it a HTTP request and evaluate the response based on the configuration to determine if the serive is alive or not. When the endpoint type is `http-proxy`, the healthcheck will reuse the same HTTP client to contact the service as the one used by a incoming gateway request. For other kind of endpoints (like kafka or mqtt), the health-check service will initiate its own HTTP client.

Reviewer: ```suggestion
The *Http health-check Service* checks the healthy state of the backend service by sending an HTTP request and evaluates the response based on the configuration to determine if the service is alive or not. When the endpoint type is `http-proxy`, the health-check will reuse the same HTTP client to contact the service as the one used by an incoming gateway request. For other kinds of endpoints (like Kafka or MQTT), the health-check service will initiate its own HTTP client.
```
